src/extensions/averagewaveforms/AverageWaveformsView/fetch_average_waveforms_2.py

```diff
@@ -38,6 +39,12 @@ def fetch_average_waveform_2(snippets_h5, unit_id):
         unit_spike_train = np.array(f.get(f'unit_spike_trains/{unit_id}'))
         unit_waveforms = np.array(f.get(f'unit_waveforms/{unit_id}/waveforms'))
         unit_waveforms_channel_ids = np.array(f.get(f'unit_waveforms/{unit_id}/channel_ids'))
+
+        if visible_channel_ids is not None:
```

Comment:
Is getting the average waveforms an expensive enough operation that this should be two separate hither calls? As is it looks to me like the way caching works, this will compute everything, but then store the result specific to the `visible_channel_ids` parameter, so that the same set of average waveforms could be computed O(n!) times if you had someone going through all the possible combinations of channel ids.
OTOH people are unlikely to do that, and if the computation is cheap, it doesn't matter anyway.

Reply:
you are right. I think I did this the wrong way. The visible_channel_ids should not be a parameter here... it should be on the client side. I did it the right way for snippets. Opening an issue.